src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -77,11 +77,21 @@
     Optional<Course> findById(long courseId);
 
     @Query("""
-            select c
-            from Course c, User u
-            where u.id = :#{#userId} and c.instructorGroupName member of u.groups
+            select distinct c
+            from Course c left join fetch c.exercises e, User u
+            where u.id = :#{#userId} and c.instructorGroupName member of u.groups and
+            exists (select exercise from Exercise exercise
+                where exercise.course.id = c.id and TYPE(exercise) = QuizExercise)
```

Comment:
I think this can be done with join, avoiding the nested query.

Reply:
👍